core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/progress/listener/ProgressListenerInvoker.java

```diff
@@ -0,0 +1,110 @@
+/*
```

Comment:
We dont have defined testing for this class on its own. In a subsequent PR defining a ProgressUpdater class through which this class is accessed, I will define Unit Tests to test the E2E functioning